src/ol/render/canvas/style.js

```diff
@@ -298,11 +298,16 @@ export function buildStyle(flatStyle, context) {
  * @return {FillEvaluator?} A function that evaluates to a fill.
  */
 function buildFill(flatStyle, prefix, context) {
-  const evaluateColor = colorLikeEvaluator(
-    flatStyle,
-    prefix + 'fill-color',
-    context
-  );
+  let evaluateColor;
+  if ('fill-pattern-src' in flatStyle) {
```

Comment:
This fixes the error and the artifacts that appear when combining pattern and text:

```suggestion
  if (
    prefix !== 'text-' &&
    prefix !== 'text-background-' &&
    'fill-pattern-src' in flatStyle
  ) {
```

![image](https://github.com/openlayers/openlayers/assets/8948865/0cc38789-59a0-49ec-806e-bc132e782284)